### PR TITLE
fix: type.attributes.map is not a function & batch translation  

### PR DESCRIPTION
**Pull request by breezertwo**
* **fix: type.attributes.map is not a function** (#526) 
This [comment](https://github.com/Fekide/strapi-plugin-translate/issues/404#issuecomment-3362494435) mentioned the root of the error. The reducer in `addTranslationToFields` uses `{}` as inital value. Due to typing issues I could not replace the object with an empty array. Using `Objec.values` and `map` instead returns the expected array. 

* **fix: batch translation update fails due to empty relations** (#532)
For batch translation we always got an error that the translation failed, but we where seeing that tokens had been used on deepl. 
`translateEntity` from the translate service threw the error and I could trace it back to `strapi.documents(params.contentType).update` in this method, which was throwing due to some entries in the `cleanedData` object where `relationName: []`. I added a fix to delete all empty arrays from the object before returning it. With this fix, all updates where working correct again. 

ℹ️ 
As our content team was running into these issues with the beta I attempted a fix which is working fine for us. Maybe the fixes can be released as a update for the current `next` version so others can make use of it, if the fixes are fine and approved. At least somebody may use my fixes as a reference if the need to patch it them selfs. 
Let me know if you need any more infos here or feel free to close this PR if is not of use. 
Thanks for this awesome plugin.

### CODE REVIEW

File: plugin/server/src/utils/clean-data.ts [hunk 1]
```diff
@@ -83,6 +83,14 @@ export function cleanData<
...
     }
   })
+
+  // Remove empty arrays after all processing
+  Object.keys(resultData).forEach((key) => {
+    if (Array.isArray(resultData[key]) && resultData[key].length === 0) {
+      delete resultData[key]
+    }
+  })
+
```
Comment (sargreal) [inline plugin/server/src/utils/clean-data.ts:91]: This should be moved inside the loop over all attributes, just after all of the other processing.

This would also fix the failing unit test about when  empty components are passed

**Review by sargreal — CHANGES_REQUESTED** [review]
Thank you @breezertwo for trying to fix those issues for the new strapi versions!

I believe this is almost ready, just one small fix, related to the unit tests.

File: plugin/server/src/utils/clean-data.ts (hunk 1 above)
Comment (breezertwo) [reply plugin/server/src/utils/clean-data.ts:91]: @sargreal Moved the deletion inside the `forEach` loop. Locally the unit tests are green now 👍

**Review by sargreal — APPROVED** [review]
Thanks a lot for this!

And sorry for the long wait
